src/PostgREST/DbStructure.hs

```diff
@@ -101,15 +101,33 @@ getDbStructure schemas extraSearchPath pgVer prepared = do
       cols' = addForeignKeys rels cols
       keys' = addViewPrimaryKeys srcCols keys
 
-  return DbStructure {
+  return $ removeInternal schemas $ DbStructure {
       dbTables = tabs
     , dbColumns = cols'
-    , dbRelations = rels
+    , dbRelationships = rels
     , dbPrimaryKeys = keys'
     , dbProcs = procs
     , pgVersion = pgVer
     }
 
+-- | Remove db objects that belong to an internal schema(not exposed through the API) from the DbStructure.
+removeInternal :: [Schema] -> DbStructure -> DbStructure
+removeInternal schemas dbStruct =
```

Comment:
Hm... I'm not a big fan of filtering this in haskell. But don't see a way to do this on the SQL side already - short of refactoring a lot of the queries in here. Reminds of the the megaquery approach!

Reply:
Yeah, IIRC getting the relationships with one big query is possible. But it's a big change, not only the queries need to be changed but also the types/functions. This is definitely the simplest way(not that bad I think, just one function).